### PR TITLE
(BKR-391) Add fqdn to /etc/hosts

### PR DESCRIPTION
**Pull request by mcanevet**

### CODE REVIEW

**Comment by anodelman** [comment]
ok to test

**Comment by anodelman** [comment]
Fails spec:
```
Failures:

  1) Beaker::Docker#provision should generate a new /etc/hosts file referencing each host
     Failure/Error: docker.hack_etc_hosts( hosts, options )
       #<Beaker::Docker:0x000000071ddc68> received :set_etc_hosts with unexpected arguments
         expected: (vm1, "127.0.0.1\tlocalhost localhost.localdomain\n192.0.2.1\tvm1\n192.0.2.1\tvm2\n192.0.2.1\tvm3\n")
              got: (vm1, "127.0.0.1\tlocalhost localhost.localdomain\n192.0.2.1\tvm1. vm1\n192.0.2.1\tvm2. vm2\n192.0.2.1\tvm3. vm3\n")
       Diff:
       @@ -1,3 +1,3 @@
        [vm1,
       - "127.0.0.1\tlocalhost localhost.localdomain\n192.0.2.1\tvm1\n192.0.2.1\tvm2\n192.0.2.1\tvm3\n"]
       + "127.0.0.1\tlocalhost localhost.localdomain\n192.0.2.1\tvm1. vm1\n192.0.2.1\tvm2. vm2\n192.0.2.1\tvm3. vm3\n"]
       
     # ./lib/beaker/host_prebuilt_steps.rb:343:in `block in hack_etc_hosts'
     # ./lib/beaker/host_prebuilt_steps.rb:342:in `each'
     # ./lib/beaker/host_prebuilt_steps.rb:342:in `hack_etc_hosts'
     # ./spec/beaker/hypervisor/docker_spec.rb:210:in `block (3 levels) in <module:Beaker>'

  2) Beaker::Vagrant can generate a new /etc/hosts file referencing each host
     Failure/Error: vagrant.hack_etc_hosts( @hosts, options )
       #<Beaker::Vagrant:0x000000072c5a90> received :set_etc_hosts with unexpected arguments
         expected: (vm1, "127.0.0.1\tlocalhost localhost.localdomain\nip.address.for.vm1\tvm1\nip.address.for.vm2\tvm2\nip.address.for.vm3\tvm3\n")
              got: (vm1, "127.0.0.1\tlocalhost localhost.localdomain\nip.address.for.vm1\tvm1. vm1\nip.address.for.vm2\tvm2. vm2\nip.address.for.vm3\tvm3. vm3\n")
       Diff:
       @@ -1,3 +1,3 @@
        [vm1,
       - "127.0.0.1\tlocalhost localhost.localdomain\nip.address.for.vm1\tvm1\nip.address.for.vm2\tvm2\nip.address.for.vm3\tvm3\n"]
       + "127.0.0.1\tlocalhost localhost.localdomain\nip.address.for.vm1\tvm1. vm1\nip.address.for.vm2\tvm2. vm2\nip.address.for.vm3\tvm3. vm3\n"]
       
     # ./lib/beaker/host_prebuilt_steps.rb:343:in `block in hack_etc_hosts'
     # ./lib/beaker/host_prebuilt_steps.rb:342:in `each'
     # ./lib/beaker/host_prebuilt_steps.rb:342:in `hack_etc_hosts'
     # ./spec/beaker/hypervisor/vagrant_spec.rb:135:in `block (2 levels) in <module:Beaker>'
```

**Comment by mcanevet** [comment]
@anodelman I think I'll need a little help to mock the Command call here https://github.com/puppetlabs/beaker/blob/308cb67213abadc35d324e9b6c5217d09743c594/lib/beaker/host_prebuilt_steps.rb#L275

**Comment by mcanevet** [comment]
@anodelman unit tests fixed

**Comment by mcanevet** [comment]
@anodelman well this is weird, it works for me if I run `rake spec`

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
retest this please

File: lib/beaker/host_prebuilt_steps.rb [hunk 1]
```diff
@@ -336,7 +336,8 @@ def copy_ssh_to_root host, opts
     def hack_etc_hosts hosts, opts
       etc_hosts = "127.0.0.1\tlocalhost localhost.localdomain\n"
       hosts.each do |host|
-        etc_hosts += "#{host['vm_ip'] || host['ip'].to_s}\t#{host[:vmhostname] || host.name}\n"
+        domain = get_domain_name(host)
+        etc_hosts += "#{host['vm_ip'] || host['ip'].to_s}\t#{host[:vmhostname] || host.name}.#{domain} #{host[:vmhostname] || host.name}\n"
```
Comment (rick) [inline lib/beaker/host_prebuilt_steps.rb]: Maybe in the process I would go a step further and improve readability here a notch:

``` ruby
  ip = host['vm_ip'] || host['ip'].to_s
  hostname = host[:vmhostname] || host.name
  etc_hosts += "#{ip}\t#{hostname}.#{domain} #{hostname}\n"
```

**Comment by anodelman** [comment]
retest this please

**Pull request merged**